src/Component.js

```diff
@@ -117,7 +130,7 @@ export class DebounceInput extends React.PureComponent {
   };
 
 
-  createNotifier = debounceTimeout => {
+  createNotifier = (debounceTimeout, debounceOptions) => {
```

Comment:
Sorry for a nit, but I usually do named arguments if there are more then one. Like `({debounceTimeout, debounceOptions})`

Reply:
These are named arguments: `(debounceTimeout, debounceOptions) => ` and this is destructuring named properties from the first argument:  `({debounceTimeout, debounce}) => `. If you change it to the latter, then every call to `createNotifier` will have to create a new object (unless of course you have a single object instance that you keep around to pass to it) e.g. `createNotifier({debounceTimeout: theTimeout, debounceOptions: theOptions})` (which of course could be shortened if the variables are named the same as the properties).

Reply:
Yeah, right. Destructuring in JS is the only way of doing "named arguments" thing, which just does not exist in JS natively. Gives more clarity to the code that calls this sort of function.